### PR TITLE
Windows: Compilation enablement

### PR DESCRIPTION
**Pull request by jhowardmsft**
Signed-off-by: John Howard <jhoward@microsoft.com>

This factors a bunch of things to allow Windows to compile. There remains much work to actually make it do something though. Combination of #49, #52, #53 and this will allow successful compilation.

### CODE REVIEW

**Comment by jhowardmsft** [comment]
I need to still verify Linux compilation - wouldn't be surprised if there are imports missing or unused. Will do 6/28.

File: checkpoint_windows.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+import (
+	"github.com/codegangsta/cli"
+)
+
+var checkpointCommand = cli.Command{}
+
+//func getCheckpointImagePath(context *cli.Context) string {
+//return ""
+//}
```
Comment (thaJeztah) [inline checkpoint_windows.go]: Guess this should be removed, or a specific reason to leave this in?

Comment (jhowardmsft) [inline checkpoint_windows.go]: Yeah, was a hang-over from when I was trying to get this to compile (Windows won't support checkpoints). Removed.

**Comment by LK4D4** [comment]
@jhowardmsft Need rebase.

**Comment by crosbymichael** [comment]
Right now lets not checkin dead code or a bunch of methods that return nil and do nothing else.  

we still have alot of figure out so lets take is slow until we have a draft spec'd for both linux and windows before changing code at the moment.

**Comment by jhowardmsft** [comment]
@crosbymichael The problem I have is if I can't even get a runc which compiles on Windows, it makes it extremely difficult to put the actual working code in there piece by piece. Hence why the four submitted to date are critical to avoid huge PRs in the future. Exactly the same as the bring-up of the Windows daemon....

**Pull request closed without merging**